pvlib/ivtools.py

```diff
@@ -0,0 +1,268 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Mar 29 10:34:10 2019
+
+@author: cwhanse
+"""
+
+import numpy as np
+
+
+def fit_cec_sam(celltype, v_mp, i_mp, v_oc, i_sc, alpha_sc, beta_voc,
+                gamma_pmp, cells_in_series, temp_ref=25):
+    '''
+    Estimates parameters for the CEC single diode model [1] using the SAM SDK.
+
+    Parameters
+    ----------
+    celltype : str
+        Value is one of 'monoSi', 'multiSi', 'polySi', 'cis', 'cigs', 'cdte',
+        'amorphous'
+    v_mp : float
+        Voltage at maximum power point at standard test condition (STC) [V]
+    i_mp : float
+        Current at maximum power point at STC [A]
+    v_oc : float
+        Open circuit voltage at STC [V]
+    i_sc : float
+        Short circuit current at STC [A]
+    alpha_sc : float
+        Temperature coefficient of short circuit current at STC [A/C]
+    beta_voc : float
+        Temperature coefficient of open circuit voltage at STC [V/C]
+    gamma_pmp : float
+        Temperature coefficient of power at maximum point point at STC [%/C]
+    cells_in_series : int
+        Number of cells in series
+    temp_ref : float, default 25C
+        Reference temperature condition
+
+    Returns
+    -------
+    a_ref : float
+        The product of the usual diode ideality factor (n, unitless),
+        number of cells in series (Ns), and cell thermal voltage at reference
+        conditions, in units of V.
+
+    I_L_ref : float
+        The light-generated current (or photocurrent) at reference conditions,
+        in amperes.
+
+    I_o_ref : float
+        The dark or diode reverse saturation current at reference conditions,
+        in amperes.
+
+    R_sh_ref : float
+        The shunt resistance at reference conditions, in ohms.
+
+    R_s : float
+        The series resistance at reference conditions, in ohms.
+
+    Adjust : float
+        The adjustment to the temperature coefficient for short circuit
+        current, in percent.
+
+    Raises:
+        ImportError if NREL-PySAM is not installed
+
+    References
+    ----------
+    [1] A. Dobos, "An Improved Coefficient Calculator for the California
+    Energy Commission 6 Parameter Photovoltaic Module Model", Journal of
+    Solar Energy Engineering, vol 134, 2012.
+    '''
+
+    try:
+        from PySAM import PySSC
+    except ImportError as e:
+        raise(e)
+
+    datadict = {'tech_model': '6parsolve', 'financial_model': 'none',
+                'celltype': celltype, 'Vmp': v_mp,
+                'Imp': i_mp, 'Voc': v_oc, 'Isc': i_sc, 'alpha_isc': alpha_sc,
+                'beta_voc': beta_voc, 'gamma_pmp': gamma_pmp,
+                'Nser': cells_in_series, 'Tref': temp_ref}
+
+    result = PySSC.ssc_sim_from_dict(datadict)
```

Comment:
Need to check for fitting success before assigning results, because dictionary entries are apparently missing altogether when it fails causing a `KeyError` at assignment:
```python
if result['cmod_success'] == 1:
    # Assign everything
else:
    # Assign nan's like in fit_sde_sandia()?
```
This behavior also needs to be tested and documented.

Here's how I broke it:
```python
v_mp = 0.45
i_mp = 5.25
v_oc = 0.55
i_sc = 5.5
alpha_sc = 0.0005 * i_sc
beta_voc = 0.005 * v_oc
gamma_pmp = 0.0055
cells_in_series = 1
temp_ref = 25
for celltype in ['monoSi', 'multiSi', 'polySi', 'cis', 'cigs', 'cdte', 'amorphous']:
    ivtools.fit_cec_sam(celltype, v_mp, i_mp, v_oc, i_sc, alpha_sc, beta_voc, gamma_pmp, cells_in_series, temp_ref)
```

Reply:
Thanks!